### ISSUE TITLE
[dicussion] Exporter Module

### ISSUE THREAD

**Issue by vabene1111** [hhursev/recipe-scrapers#619]
i am not sure if this is on purpose left out of the application so far but given that I write functions like those when developing a scraper anyway I was wondering if you guys would like to introduce a small set of utility functions that allow a person who scraps a recipe to quickly "view" what the scraper can return about the recipe. 

This would help people understand what a recipe request returned and also help with developing new and testing old scrapers. 

I was thinking of a `json` and a `markdown` module that simply returns a super basic structure of a recipe in a somewhat human readable way so you can easily and quickly view a recipe. If people like it it could be expanded to things like html, pdf or whatever people like.

If you say its out of scope for this project I totally understand it, if you would like this I would provide a draft of how I would implement this and take a look at implementing if we can agree on a solution.

**Comment by jayaddison** [comment]
JSON export would be excellent.  As a user, I think the way I'd expect to do that in Python would be:

```
import json
from recipe_scrapers import scrape_me

recipe = scrape_me("https://.../example/recipe.html")
print(json.dumps(recipe, indent=True))
```

... but that doesn't work yet, because our scraper objects aren't JSON serializable.

Does that pattern for exporting recipes look like a good interface?  Would that be a good first goal for something like this?

**Comment by vabene1111** [comment]
yes and no

yes because this is the way a normal python user would expect a json serialization to work and its very easy to use and probably also implement. 

on the other hand this would mean that the syntax for exporting markdown, pdf, html or anything else would need to be different as there are no standard serialization methods for that. 

the format that I had in mind was something like (very rough)

```
from recipe_scrapers import scrape_me, Exporter

recipe = scrape_me("https://.../example/recipe.html")

exporter = Exporter(recipe)
exporter.to_json()
exporter.to_markdown()
```

**Comment by jayaddison** [comment]
Ok, understood.  However: I'm not keen on the idea of supporting export to primarily human-readable output file formats from this library.

The main reason is that producing those would require us to implement and maintain presentation-layer decisions:

For example, we'd develop templates for selected languages/locales, and make decisions on the formatting style for the recipe fields -- those are decisions that users are likely to want to customize for themselves.

A library/tool that can do that is a good idea (it could use `recipe-scrapers` as a dependency and export recipes into many different formats), but I think that'd be better developed and maintained separately (the skills required and community have overlap, but seem slightly different).

**Comment by vabene1111** [comment]
Ok perfectly fine, that's why I asked if something like this would be of interest to the library.

I think it would be doable with a very basic markup but json only is fine too. If I find some time I will try to write a json dump function

**Comment by jayaddison** [comment]
An idea related to this: if we export JSON in `schema.org` format ([`Recipe`](https://schema.org/Recipe) type), then perhaps we could include a link to the URL we retrieved it from by exporting a [`translationOfWork`](https://schema.org/translationOfWork) field.

**Comment by bfcarpio** [comment]
I know you linked the USDA PR, but I think some of this would be solved with the dataclass discussion. JSON serialization is supported natively or very thoroughly by the community and I'm sure there are other packages that can turn them into parquet files or what have you.

They might be a good way to create an interface between data and presentation.

**Comment by jayaddison** [comment]
Resolved by #621 (I think; correct me if I'm wrong).

**Comment by vabene1111** [comment]
yes that would be the case, still some improvements possible but a good solution for now.
